docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -19,33 +19,36 @@ Each project stores blocks in different places and uses slightly different block
 
 For configuration of remote write to Grafana Mimir, refer to [Configuring Prometheus remote write]({{< relref "../operators-guide/secure/authentication-and-authorization.md#configuring-prometheus-remote-write" >}}).
 
-## Uploading historic blocks to the Grafana Mimir storage bucket
+## Uploading historic TSDB blocks to Grafana Mimir
+
+Grafana Mimir supports uploading of historic TSDB blocks, notably from Prometheus.
+In order to enable this functionality, either for all tenants or a specific one, refer to
+[Configuring TSDB block upload]({{< relref "../operators-guide/configure/configure-tsdb-block-upload.md" >}}).
 
 Prometheus stores TSDB blocks in the path specified in the `--storage.tsdb.path` flag.
 
-To find all blocks directories in the TSDB `<STORAGE TSDB PATH>`, run the following command:
+To find all block directories in the TSDB `<STORAGE TSDB PATH>`, run the following command:
 
 ```bash
 find <STORAGE TSDB PATH> -name chunks -exec dirname {} \;
 ```
 
-Grafana Mimir supports multiple tenants and stores blocks with a tenant prefix.
-With multi-tenancy disabled, there is a single tenant called `anonymous`.
-
-Copy each directory output from the previous command to the Grafana Mimir object storage bucket with
-your tenant prefix.
+Grafana Mimir supports multiple tenants and stores blocks per tenant. With multi-tenancy disabled, there
+is a single tenant called `anonymous`.
 
-- For AWS S3, use the `aws` tool in the following command:
+Use Grafana mimirtool to upload each block, such as those identified by the previous command, to Grafana Mimir:
 
 ```bash
-aws s3 cp <DIRECTORY> s3://<TENANT>/<DIRECTORY>
+mimirtool backfill --address=http://<mimir-hostname> --id=<tenant> <block1> <block2>...
 ```
 
-- For Google Cloud Storage (GCS), use the `gsutil` tool in the following command:
+> **Note**: If you need to authenticate against Grafana Mimir, you can provide an API key via the `--key` flag,
+> for example `--key=$(cat token.txt)`.
 
-```bash
-gsutil -m cp -r <DIRECTORY> gs://<TENANT>/<DIRECTORY>
-```
+Grafana Mimir performs some sanitization and validation of each block's metadata.
+As a result, it rejects Thanos blocks due to unsupported labels.
+As a workaround, if you need to upload Thanos blocks, upload the blocks directly to the
+Grafana Mimir blocks bucket, prefixed by `<tenant>/<block ID>/`.
```

Comment:
I disagree on this. There's a reason why we reject Thanos blocks with external labels. We shouldn't recommend to just copy-paste them in the storage. Instead, we should link to "Considerations on Thanos specific features" below.

Reply:
Thank you @pracucci :)

Reply:
We have entire section dedicated to Thanos blocks, and workarounds to get them working. If we completely discourage people from uploading Thanos blocks to Mimir storage, we can just say it's not supported, and remove all that text.

Reply:
> We have entire section dedicated to Thanos blocks, and workarounds to get them working. If we completely discourage people from uploading Thanos blocks to Mimir storage, we can just say it's not supported, and remove all that text.

@pracucci what's the consensus?